File: python/MaximizingXOR.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the maximizingXor function below.
def maximizingXor(l, r):
    max_value = 0
    for i in range(l, r+1):
        for j in range(l, r+1):
            if max_value < i ^ j:
                max_value = i ^ j
    
    return max_value

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    l = int(input())

    r = int(input())

    result = maximizingXor(l, r)

    fptr.write(str(result) + '\n')

    fptr.close()

File: scala/MaximizingXOR.scala
import java.io._
import java.math._
import java.security._
import java.text._
import java.util._
import java.util.concurrent._
import java.util.function._
import java.util.regex._
import java.util.stream._

object Solution {

    // Complete the maximizingXor function below.
    def maximizingXor(l: Int, r: Int): Int = {
        var max_value = 0
        for (i <- l to r)
            for (j <- l to r)
                if (max_value < (i^j))
                    max_value = i^j
    
        return max_value

    }

    def main(args: Array[String]) {
        val stdin = scala.io.StdIn

        val printWriter = new PrintWriter(sys.env("OUTPUT_PATH"))

        val l = stdin.readLine.trim.toInt

        val r = stdin.readLine.trim.toInt

        val result = maximizingXor(l, r)

        printWriter.println(result)

        printWriter.close()
    }
}
